storage/posix/files.go

```diff
@@ -401,16 +438,58 @@ func (s *Storage) initialise(create bool) error {
 }
 
 // writeCheckpoint stores a raw log checkpoint on disk.
-func writeCheckpoint(path string, newCPRaw []byte) error {
-	if err := createExclusive(filepath.Join(path, layout.CheckpointPath), newCPRaw); err != nil {
-		return fmt.Errorf("failed to create checkpoint file: %w", err)
+func (s *Storage) writeCheckpoint(newCPRaw []byte) error {
+	if err := createExclusive(filepath.Join(s.path, stateDir, layout.CheckpointPath), newCPRaw); err != nil {
+		return fmt.Errorf("failed to create private checkpoint file: %w", err)
+	}
+	// Notify that we know for sure there's a new checkpoint, but don't block if there's already
+	// an outstanding notification in the channel.
+	select {
+	case s.cpUpdated <- struct{}{}:
+	default:
 	}
 	return nil
 }
 
-// readcheckpoint returns the latest stored checkpoint.
-func readCheckpoint(path string) ([]byte, error) {
-	return os.ReadFile(filepath.Join(path, layout.CheckpointPath))
+// readcheckpoint returns the latest stored private checkpoint.
+func (s *Storage) readCheckpoint() ([]byte, error) {
+	return os.ReadFile(filepath.Join(s.path, stateDir, layout.CheckpointPath))
+}
+
+// publishCheckpoint makes the most recently checkpoint from the stateDir available at the
+// (public) tlog-tiles specified Checkpoint location.
+func (s *Storage) publishCheckpoint(minStaleness time.Duration) error {
+	// Lock the destination "published" checkpoint location:
+	cpF, err := lockFile(filepath.Join(s.path, layout.CheckpointPath))
+	if err != nil {
+		return fmt.Errorf("lockFile(%s): %v", layout.CheckpointPath, err)
+	}
+	defer func() {
+		if err := cpF.Close(); err != nil {
```

Comment:
Should this call `unlockFile` instead of close? Or just get rid of the unlockFile method because it's an alias...

Reply:
It should call `unlockFile`, this was an abstraction "leak" - fixed.